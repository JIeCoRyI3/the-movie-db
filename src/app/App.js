import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import styles from './App.module.css';
import HomePage from '../components/HomePage';
import NotFoundPage from '../components/NotFoundPage';
import Footer from '../components/Footer';
import { connect } from 'react-redux';
import { loadData } from './connect-store';
import MainContainer from '../components/MainContainer';
import LoaderComponent from '../components/LoaderComponent';

class App extends React.Component {
    componentDidMount() {
        this.props.loadData();
    }

    render() {
        return this.props.genres ? (
            <BrowserRouter>
                <div className={styles.App}>
                    <Switch>
                        <Route path={'/'} component={HomePage} exact />
                        <Route path={'/film/:id'} component={HomePage} exact />
                        <Route path={'*'} component={NotFoundPage} />
                    </Switch>
                    <MainContainer />
                    <Footer />
                </div>
            </BrowserRouter>
        ) : (
            <LoaderComponent />
        );
    }
}

const mapStateToProps = (state) => ({
    genres: state.app.genresList,
});

const mapDispatchToProps = {
    loadData,
};

const withStore = connect(mapStateToProps, mapDispatchToProps);

export default withStore(App);
