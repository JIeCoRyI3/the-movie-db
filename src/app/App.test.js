import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('<App />', () => {
    it('should could load data on componetDidMount', () => {
        const props = {
            loadData: jest.fn(() => {}),
        };
        const wrapper = shallow(<App {...props} />);
        const instance = wrapper.instance();

        instance.componentDidMount();
        expect(props.loadData).toHaveBeenCalled();
    });
});
