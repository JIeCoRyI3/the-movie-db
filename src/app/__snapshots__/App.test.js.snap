// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should something 1`] = `ShallowWrapper {}`;
